core/domain/acl_decorators.py

```diff
@@ -126,3 +167,320 @@ def test_can_view_stats(self, exploration_id, **kwargs):
             raise self.PageNotFoundException
 
     return test_can_view_stats
+
+
+def can_create_exploration(handler):
+    """Decorator to check whether the user can create an exploration."""
+
+    def test_can_create(self, **kwargs):
+        if (
+                (role_services.ACTION_CREATE_EXPLORATION in self.actions) and
+                user_services.has_fully_registered(self.user_id)):
+            return handler(self, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to create an exploration.')
+
+    return test_can_create
+
+
+def can_create_collection(handler):
+    """Decorator to check whether the user can create a collection."""
+
+    def test_can_create(self, **kwargs):
+        if (
+                (role_services.ACTION_CREATE_COLLECTION in self.actions) and
+                user_services.has_fully_registered(self.user_id)):
+            return handler(self, **kwargs)
+        else:
+            raise self.UnauthorizedUserException(
+                'You do not have credentials to a create collection.')
+
+    return test_can_create
+
+
+def can_access_creator_dashboard(handler):
+    """Decorator to check whether the user can access creator dashboard
+    page.
+    """
+
+    def test_can_access(self, **kwargs):
+        if self.user_id is None:
+            self.redirect(utils.set_url_query_parameter(
+                feconf.SIGNUP_URL, 'return_url',
+                '/notifications_dashboard'))
```

Comment:
For a logged-out user, we want the following sequence of actions to work: go to splash page and click create button, which takes them to a log in screen, which takes them to a registration/signup screen, which takes them to the creator dashboard. What I'm asking is whether this flow still works? Otherwise we might need to change some stuff in the frontend to make sure the correctly-formed URLs are being called (so that we don't trust to server redirection).